pkg/scheduler/scheduler.go

```diff
@@ -36,6 +37,38 @@ type pollResult struct {
 	errs     []error
 }
 
+// healthCheckHandler is a simple health check handler for the HTTP server.
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	// Define a simple JSON response structure
+	response := struct {
+		Status  string `json:"status"`
+		Message string `json:"message"`
+	}{
+		Status:  "OK",
+		Message: "Service is up and running",
+	}
+
+	// Serialize the response structure to JSON
+	jsonResponse, err := json.Marshal(response)
+	if err != nil {
+		// If there's an error, log it and return a 500 Internal Server Error
+		log.Errorf("Failed to marshal health check response: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Write an HTTP status code to the response
+	w.WriteHeader(http.StatusOK)
+
+	// Write the JSON response body
+	_, err = w.Write(jsonResponse)
+	if err != nil {
+		log.Errorf("Failed to write health check response: %v", err)
```

Comment:
Should we just silently log the error? It seems to me like it would be kind of problematic if we're not able to write the healthcheck response. That could be indicative of other issues on the server.